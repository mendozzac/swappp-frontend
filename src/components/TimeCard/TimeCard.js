const TimeCard = ({ time }) => {
  return (
    <li className="time">
      <div>
        <span>{time.distance}</span>
        <span>{time.style}</span>
        <span>{time.pool}</span>
      </div>
    </li>
  );
};

export default TimeCard;
