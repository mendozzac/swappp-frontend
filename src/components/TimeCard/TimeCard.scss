@import "../../styles/colors";

.time {
  margin: 10px;
  &-box {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
  }
  &-element {
    color: $darkgray;
  }
}

@media (min-width: 1000px) {
  .time-box {
    text-align: center;
    font-size: 20px;
  }
}
