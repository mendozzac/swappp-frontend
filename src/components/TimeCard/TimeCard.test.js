import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { MemoryRouter } from "react-router-dom";
import ReactTestRenderer from "react-test-renderer";
import TimeCard from "./TimeCard";

describe("Given a TimeCard component", () => {
  describe("When it receives time information", () => {
    test("Then it should render a card with the information of the time", () => {
      const time = { meters: "200", style: "Braza", time: "2:16.02" };

      render(
        <BrowserRouter>
          <TimeCard time={time} />
        </BrowserRouter>
      );
      const newTime = screen.getByText("Braza");

      expect(newTime).toBeInTheDocument();
    });
  });
  describe("When it receives a time", () => {
    test("Then it should render a card with the time", () => {
      const newTime = {};

      const timeCard = ReactTestRenderer.create(
        <MemoryRouter>
          <TimeCard time={newTime} />
        </MemoryRouter>
      );

      expect(timeCard.toJSON()).toMatchSnapshot();
    });
  });
});
