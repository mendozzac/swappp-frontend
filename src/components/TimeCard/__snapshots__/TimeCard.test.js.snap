// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a TimeCard component When it receives a time Then it should render a card with the time 1`] = `
<li
  className="time-box"
>
  <div
    className="time"
  >
    <span
      className="time-element"
    />
  </div>
  <div
    className="time"
  >
    <span
      className="time-element"
    />
  </div>
  <div
    className="time"
  >
    <span
      className="time-element"
    />
  </div>
</li>
`;
